service/src/main/java/org/mskcc/cmo/metadb/service/impl/AdminMessageHandlingServiceImpl.java

```diff
@@ -83,37 +94,27 @@ public void run() {
-                    // TODO:
-                    // database operations
-                    //      1. find patient node matching old id and update with new id
-                    //      2. get all RESEARCH samples linked to affected patient
-                    // publish each research sample as a sample metadata json to the CMO LABEL UPDATE topic
-                    // for label generator to make a new label
                     if (idCorrectionMap != null) {
-                        // parse the old cmo patient id and new cmo patient id from map
-                        System.out.println("TODO: PARSE OLD AND NEW CMO PATIENT ID FROM REPLYINFO");
-                        // make updates to the database
-                        // return some sort of indicator that tells us if patient ids were
-                        // successfully mapped or not
-                        System.out.println("TODO: MAKE CORRECTIONS TO CMO PATIENT ID");
-                        // for each RESEARCH sample that was linked to the affected cmo patient node
-                        // we need to publish to the CMO_LABEL_UPDATE request-reply topic
-                        // and in return we receive an updated cmo label
-                        // label generator is also publishing the updated sample metadata to
-                        // IGO_SAMPLE_METADATA_UPDATE which will be handled by the existing
-                        // metadb sample metadata update subscriber that will persist the updates
-                        // to the database - we dont need to do anything with the received
-                        // label at this point other than log that a new one has been generated
-                        // and that the updated metadata will go through the appropriate
-                        // sample update message handling
-                        System.out.println("TODO: PUBLISH EACH SAMPLE TO THE CMO LABEL"
-                                + "UPDATE TOPIC");
-                        // this is a request-reply topic that also publishes to the
-                        // sample metadata update topic that is handled by metadb so
-                        // what we can do here is simply log that the label change from
-                        // "old label" --> "new label" should be persisted shortly
-                        System.out.println("TODO: LOG THAT OLD LABEL IS BEING UPDATED TO "
-                                + "NEW LABEL AND SHOULD BE PERSISTED IN DB SHORTLY");
+                        String oldCmoPatientId = idCorrectionMap.get("oldId");
+                        String newCmoPatientId = idCorrectionMap.get("newId");
+                        MetadbPatient updatedPatient = patientService.updateCmoPatientId(
+                                oldCmoPatientId, newCmoPatientId);
+                        
+                        List<SampleMetadata> sampleMetadataList = sampleService
```

Comment:
I still suggest moving this call so that it's ahead of the call to update the cmo patient id 

looking at how the query is set up in `MetadbSampleRepository.findAllSampleMetadataListByCmoPatientId()` - the list of sample metadata is fetched by matching on the patient alias nodes and not the sample node `cmoPatientId` property 

```
    @Query("MATCH (pa: PatientAlias {namespace: 'cmoId', value: $cmoPatientId})-[:IS_ALIAS]->"
            + "(p: Patient)-[:HAS_SAMPLE]->(s: Sample)-[:HAS_METADATA]->(sm: SampleMetadata) "
            + "MATCH (r: Request)-[:HAS_SAMPLE]->(s) SET sm.igoRequestId = r.igoRequestId "
            + "RETURN sm"
    )
    List<SampleMetadata> findAllSampleMetadataListByCmoPatientId(@Param("cmoPatientId") String cmoPatientId);
```

the way you have it set up currently would mean the CMO patient alias would have the update. So assuming the cmo patient id is updated at the `PatientAlias` node, if you query for a sample metadata list by the `oldCmoPatientId` you wouldn't get any hits. 

You can just swap the placements of these two calls